cql3/type_json.cc

```diff
@@ -75,6 +77,31 @@ static sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+template <std::floating_point Tf, std::integral Ti>
+static std::partial_ordering cmp(Tf x, Ti y) {
+    return x <=> y;
+}
+
+// we are not able to fit an uint64_t or a 64 bit float into a wider numeric type
```

Comment:
It took me a while to understand the purpose of this template because the comment starts with the implementation detail instead of the goal for the function. I would start the comment with something like: 
"The following function cmp()  compares a floating-point and an integer number. Because there is no "wider" type to which a 64-bit integer and 64-bit floating point can be casted to do this comparison, we need to take a more convoluted approach:"

Reply:
good point. my head was full of the IEEE754 things when working on this, and really wanted to dump them in the form of code or document. will revise the comment as you suggest.